### ISSUE TITLE
console.table

### ISSUE THREAD

**Issue by mquandalle** [oven-sh/bun#802]
### What is the problem this feature will solve?

bun `console.table` doesn't display tabular data in a table

### What is the feature you are proposing to solve the problem?

https://nodejs.org/api/console.html#consoletabletabulardata-properties

```
> console.table([{a:1},{a:2}])
┌─────────┬───┐
│ (index) │ a │
├─────────┼───┤
│    0    │ 1 │
│    1    │ 2 │
└─────────┴───┘
```

It also works in deno

**Comment by notnoahpp** [comment]
+1 
while not widely used, its perfect for debugging

**Comment by josuamanuel** [comment]
@notnoahpp @mquandalle Though no ideal, in the mean time you have this alternative implementation compatible with bun:

npm install js-awe

```javascript
import { consoleTable } from 'js-awe'

consoleTable([{a:1},{a:2,b:2}])
```

```plaintext
┌───┬───┐
│ a │ b │
├───┼───┤
│ 1 │   │
│ 2 │ 2 │
└───┴───┘
```

 Same as console.table without the index column. If you want to add index column you can do it yourself:
 
 ```javascript
import { consoleTable } from 'js-awe'

const toConsole = [{a:1},{a:2,b:2}]
const toConsoleWithIndex = toConsole.map((el, index) => ({index, ...el}))
consoleTable(toConsoleWithIndex)
```

```plaintext
┌───────┬───┬───┐
│ index │ a │ b │
├───────┼───┼───┤
│ 0     │ 1 │   │
│ 1     │ 2 │ 2 │
└───────┴───┴───┘
```

**Comment by truongan07** [comment]
This feature is good in Nodejs but not in Bun. I need to make document in terminal. Support native API, please !

https://stackoverflow.com/questions/47228227/nodejs-how-to-display-table-structured-in-console

**Comment by PRossetti** [comment]
I would really like bun.sh to natively support something like `console.table`.

**Comment by mohamedmansour** [comment]
Bun doesn't print for `console.table` but it still works under the hood.

 Here is packageless (shim) way to do this, inspired from https://stackoverflow.com/questions/63737829/can-i-get-the-string-value-of-console-table-in-nodejs/67859384#67859384:

```
import { Console } from 'console'
import { Transform } from 'stream'

const ts = new Transform({ transform(chunk, _, cb) { cb(null, chunk) } })
const logger = new Console({ stdout: ts })

function getTable (data: any) {
  logger.table(data)
  const table = (ts.read() || '').toString()
  console.log(table)
}
console.table = getTable

**Comment by mbledkowski** [comment]
Please add this. Without this it is hard to read a large amount of data.

**Comment by andromidasj** [comment]
I agree. In [Bun's docs](https://bun.sh/docs/runtime/nodejs-apis#console), it states that `console` is fully implemented, which is not the case since `console.table` does not print a table. Would love to have this be built-in!

**Comment by truongan07** [comment]
Anyone knows to get width of terminal ?

**Comment by mohamedmansour** [comment]
> Anyone knows to get width of terminal ?

Did you try 'process.stdout.columns'

**Comment by truongan07** [comment]
> > Anyone knows to get width of terminal ?
> 
> Did you try 'process.stdout.columns'

Yet. Post all of code to get width of terminal please!

**Comment by nshiab** [comment]
It would be beneficial. But I understand it might be a low priority. Fingers crossed. 🤞

**Comment by andreasrossa** [comment]
Would also love to have this in Bun, great for CLI applications

**Comment by pjlunardelli** [comment]
This feature would be great

**Comment by Jarred-Sumner** [comment]
This was implemented by @otgerrogla in #7866 and will ship in Bun v1.0.21.
